src/frontend/src/components/Wizard/WizardProgress.tsx

```diff
@@ -0,0 +1,55 @@
+import React from "react";
+import type { VariantProps } from "tailwind-variants";
+
+import {
+    progressDotStyle,
+    progressLineStyle,
+    wizardStyle,
+} from "./Wizard.styles";
+
+interface WizardProgressProps {
+    totalSteps: number;
+    currentStep: number;
+}
+
+const getProgressInfo = (index: number, currentStep: number) => {
+    const step = index + 1;
+    const isCompleted = step < currentStep;
+    const isCurrent = step === currentStep;
+
+    let state: VariantProps<typeof progressDotStyle>["state"] = "upcoming";
+    if (isCompleted) state = "completed";
+    if (isCurrent) state = "current";
+    return { step, isCompleted, state };
+};
+
+const WizardProgress = ({ totalSteps, currentStep }: WizardProgressProps) => {
+    const { progressContainer } = wizardStyle();
+
+    return (
+        <div className={progressContainer()}>
+            {Array.from({ length: totalSteps }).map((_, index) => {
+                const { step, isCompleted, state } = getProgressInfo(
+                    index,
+                    currentStep
+                );
+                return (
+                    <React.Fragment key={step}>
+                        <div className={progressDotStyle({ state })} />
```

Comment:
We could quickly get those little checkmarks in by passing the icon as a child to the dot element:
```suggestion
                       <div className={progressDotStyle({ state })}>
                            {isCompleted && (
                                <CheckIcon className="w-[10px] h-[10px] fill-white" />
                            )}
                        </div>
```
What do you think?
<img width="522" height="625" alt="Screenshot 2025-10-08 at 1 56 37 PM" src="https://github.com/user-attachments/assets/a1f2a8ef-bc39-474e-9304-830ea71ce59e" />